From 9628600bbbeffc4ec6dc48943ee091a5d3eca18b Mon Sep 17 00:00:00 2001
From: Gavin Shan <gwshan@linux.vnet.ibm.com>
Date: Wed, 15 Feb 2017 10:53:45 +1100
Subject: lspci: Support GEN4 speed (16GT/s)

This enables "lspci" to show GEN4 speed (16GT/s) properly according
to the contents in register PCI_EXP_LNKCAP, PCI_EXP_LNKSTA and
PCI_EXP_LNKCTL2.

Reported-by: Carol Soto <clsoto@us.ibm.com>
Signed-off-by: Gavin Shan <gwshan@linux.vnet.ibm.com>
---
 ls-caps.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ls-caps.c b/ls-caps.c
index 7ff6c67..d365324 100644
--- a/ls-caps.c
+++ b/ls-caps.c
@@ -726,6 +726,8 @@ static char *link_speed(int speed)
 	return "5GT/s";
       case 3:
 	return "8GT/s";
+      case 4:
+        return "16GT/s";
       default:
 	return "unknown";
     }
@@ -1000,6 +1002,8 @@ static const char *cap_express_link2_speed(int type)
 	return "5GT/s";
       case 3:
 	return "8GT/s";
+      case 4:
+        return "16GT/s";
       default:
 	return "Unknown";
     }
-- 
cgit v1.1

